internal/provider/data_network_pool.go

```diff
@@ -0,0 +1,166 @@
+		ReadContext: dataSourceNetworkPoolRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the network pool",
+			},
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the network pool",
+			},
+			"network": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The network in the network pool",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"gateway": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The gateway of the network",
+						},
+						"mask": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The subnet mask of the network",
+						},
+						"mtu": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "The MTU of the network",
+						},
+						"subnet": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The subnet of the network",
+						},
+						"type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The type of network",
+						},
+						"vlan_id": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "The VLAN ID of the network",
+						},
+						"ip_pools": {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "The IP pools associated with the network",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"start": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The starting IP address of the IP pool",
+									},
+									"end": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The ending IP address of the IP pool",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceNetworkPoolRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(*api_client.SddcManagerClient).ApiClient
+
+	name := d.Get("name").(string)
+	networkPool, err := GetNetworkPoolByName(ctx, apiClient, name)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(networkPool.ID)
+	_ = d.Set("name", networkPool.Name)
+	_ = d.Set("networks", flattenNetworks(networkPool.Networks))
+
+	return nil
+}
+
+func GetNetworkPoolByName(ctx context.Context, apiClient *client.VcfClient, name string) (*models.NetworkPool, error) {
+	params := network_pools.NewGetNetworkPoolParamsWithContext(ctx).
+		WithTimeout(constants.DefaultVcfApiCallTimeout)
+
+	networkPoolsPayload, err := apiClient.NetworkPools.GetNetworkPool(params)
+	if err != nil {
+		return nil, err
+	}
+
+	if networkPoolsPayload.Payload == nil {
+		return nil, errors.New("error: network pool not found")
```

Comment:
I don't think you need the "error:" prefix

Reply:
I'll adjust.

Reply:
Addressed in 60f13a9.